wearipedia/devices/garmin/fenix_gen.py

```diff
@@ -12,409 +13,2833 @@
+    start_date_obj = datetime.strptime(start_date, "%Y-%m-%d")
+
+    for i in range(num_days):
+        activity_start_time = start_date_obj + timedelta(days=i)
+        # Between 30 minutes and 3 hours
+        activity_duration = random.randint(30 * 60, 3 * 60 * 60)
+        distance = random.uniform(0.0, 15.0)  # Up to 15 km
+        elevation_gain = random.uniform(0.0, 500.0)  # Up to 500 meters
+        elevation_loss = random.uniform(0.0, 500.0)  # Up to 500 meters
+        # Average speed in km/h
+        avg_speed = distance / (activity_duration / 3600)
+
+        activity_entry = {
+            "activityId": random.randint(10000000, 99999999),
+            "activityName": "Random Activity",
+            "description": None,
+            "startTimeLocal": activity_start_time.strftime("%Y-%m-%d %H:%M:%S"),
+            "startTimeGMT": activity_start_time.strftime("%Y-%m-%d %H:%M:%S"),
+            "activityType": {
+                "typeId": random.randint(1, 20),
+                "typeKey": "random_activity",
+                "parentTypeId": random.randint(1, 20),
+                "isHidden": False,
+                "restricted": False,
+                "trimmable": True,
+            },
+            "eventType": {
+                "typeId": random.randint(1, 10),
+                "typeKey": "uncategorized",
+                "sortOrder": 10,
+            },
+            "comments": None,
+            "parentId": None,
+            "distance": distance,
+            "duration": activity_duration,
+            "elapsedDuration": activity_duration,
+            "movingDuration": random.randint(20 * 60, activity_duration),
+            "elevationGain": elevation_gain,
+            "elevationLoss": elevation_loss,
+            "averageSpeed": avg_speed,
+            "maxSpeed": avg_speed + random.uniform(0.0, 5.0),
+            "startLatitude": random.uniform(-90.0, 90.0),
+            "startLongitude": random.uniform(-180.0, 180.0),
+            "hasPolyline": True,
+            "ownerId": random.randint(10000000, 99999999),
+            "ownerDisplayName": "Random User",
+            "ownerFullName": "Random User",
+        }
+
+        activities_data.append(activity_entry)
+
+    return activities_data
+
+
+def get_devices_data(start_date, num_days):
+    """Generate synthetic device data for a specified number of days.
+
+    This function generates synthetic device data for a specified number of days. It simulates various device attributes
+    for multiple devices. The generated data is structured as a list of dictionaries.
+
+    :param start_date: The start date in the format "YYYY-MM-DD".
+    :type start_date: str
+    :param num_days: The number of days for which device data should be generated.
+    :type num_days: int
+    :return: A list of dictionaries containing synthetic device data for the specified number of days.
+    :rtype: List[Dict]
+    """
+    devices_data = []
+
+    for _ in range(3):
+        devices_entry = {
```

Comment:
Since devices_entry is not altered within the loop, you can probably move it outside of the loop instead (above)

Reply:
Ok, Done!